xwiki-platform-core/pom.xml

```diff
@@ -517,6 +517,11 @@
                 <new>method long org.xwiki.like.LikeManager::countUserLikes(org.xwiki.user.UserReference) throws org.xwiki.like.LikeException</new>
                 <justification>Unstable API.</justification>
               </item>
+              <item>
+                <code>java.method.removed</code>
+                <old>method boolean org.xwiki.messagestream.GroupMessageDescriptor::isEnabled(java.lang.String)</old>
+                <justification>Not a breakage. Inherits isEnable from AbstractMessageDescriptor.</justification>
```

Comment:
I'm surprised that it needs a justification, it doesn't break the API AFAIU since the method is exposed in the abstract. Maybe we should report it to revapi.

Reply:
@surli why do you say it doesn't break the API? Removing a method can always break some usage IMO (both binary and source: https://revapi.org/revapi-java/0.22.0/differences.html#_method_removed_java_method_removed).

For example imagine that there's some code using reflection on the GroupMessageDescriptor class to get the `isEnabled` method to call it. It'll result in an exception at runtime. For the source use case, maybe if some code was calling `super.isEnabled()`, idk. There are probably cases I don't see. But I'm quite confident it's not a bug and the justification is not quite correct.

Interesting BTW that java lists the binary incompatible changes that they support: https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html. See also https://www.oracle.com/java/technologies/javase/8-compatibility-guide.html

Reply:
> For example imagine that there's some code using reflection on the GroupMessageDescriptor class to get the isEnabled method to call it. It'll result in an exception at runtime.

Well it would depend if you retrieve it by using `getDeclaredMethod` or `getMethod` since the method is declared in the superclass. Still I agree this would be a possible breakage.

Reply:
@manuelleduc so here you could actually keep the method and just call `super.isEnabled()`, or you explain a bit more why you're breaking it. Personally I'm fine with this breakage since the chances are almost null that anyone is performing reflection on this one, and it was overridden just to disable the feature half implemented...